xrviz/fields.py

```diff
@@ -136,3 +157,20 @@ def set_coords(self, data, var):
 
     def ndim_matches(self, var1, var2):
         return self.data[var1].ndim == self.data[var2].ndim
+
+    def check_are_var_coords(self):
+        '''
+        Check if both x and y are in variable's coords
+        '''
+        var_coords = list(self.data[self.var].coords)
+        x = self.x.value
+        y = self.y.value
+        return True if x in var_coords and y in var_coords else False
+
+    def guess_x_y(self, var):
+        try:
+            parsed_var = self.data.metpy.parse_cf(var)
+            x, y = parsed_var.metpy.coordinates('x', 'y')
+            return [coord.name for coord in (x, y)]
```

Comment:
Special case for longitude to x-axis goes here (can be next PR). Or should we ask metpy about clarification to get this right? Seems like their function should get the order right.

Reply:
I have asked this in [Metpy-Gitter](https://gitter.im/Unidata/MetPy#). Hopefully we will get a solution.